tests/support/__init__.py

```diff
@@ -210,15 +209,20 @@ def fixturepath(relative_path):
     return tmp_path
 
 
-def outputpath(relative_path):
-    assert not os.path.isabs(relative_path)
-    tmp_path = os.path.join(TEST_OUTPUT_DIR, relative_path)
-    return tmp_path
-
-
-def temppath(relative_path, test_case, skip_clean=False):
+def temppath(relative_path, test_case, ensure_dir=False, skip_clean=False):
     assert isinstance(test_case, MigTestCase)
     tmp_path = os.path.join(TEST_OUTPUT_DIR, relative_path)
+    if ensure_dir:
+        try:
+            os.mkdir(tmp_path)
+        except FileExistsError:
+            raise AssertionError(
+                "ABORT: use of unclean output path: %s" % relative_path)
     if not skip_clean:
         test_case._cleanup_paths.add(tmp_path)
     return tmp_path
+
+
+# compatibility alias
+def cleanpath(relative_path, test_case, **kwargs):
```

Comment:
`pylint` complains that this function is redefined. I'd tend to agree but leave it as-is in merge. Feel free to follow-up if it shouldn't be so.